### ISSUE TITLE
Code passes object in URL on DELETE request, causing it to fail

### ISSUE THREAD

**Issue by lisawilliams** [ga-wdi-boston/full-stack-project#923]
Hi all!

I am having a problem passing the right parameters with my delete request. Here's the current output from console.log, followed by the code that produces this: 

<img width="1274" alt="screen shot 2017-06-12 at 9 35 14 am" src="https://user-images.githubusercontent.com/809042/27036335-89036380-4f52-11e7-84f8-06de68af3cf0.png">

Note that it produces a URL that looks like this: 

`http://localhost:4741/chores/[object%20Object]`

Which is clearly not right, so the API rejects it with a 404. 

Here's the relevant code from index.html: 

```
  <h2>Delete a Chore</h2>
  <form id="delete-chore">
    <fieldset>
    <input type="text" name="chore[id]" id="id" placeholder="Chore ID"><br>
      <input type="submit" value="Delete chore">
    </fieldset>
  </form>

```

And from events.js: 

```
const onDeleteChore = function (event) {
  event.preventDefault()
  const id = getFormFields(event.target)
  console.log('we have arrived at onDeleteChore chores/events.js. Below is console.log event.target')
  console.log(event.target)
  console.log('Below is console.log(id)')
  console.log(id)
  api.deleteChore(id)
  .then(ui.deleteChoreSuccess)
  .catch(ui.deleteChoreFailure)
}
```

api.js: 

```
const deleteChore = (id) => {
  return $.ajax({
    url: config.apiOrigins.development + '/chores/' + id,
    method: 'GET',
    headers: {
      Authorization: 'Token token=' + store.user.token
    }
  })
}

```

and ui.js: 

```
const deleteChoreSuccess = (response) => {
  console.log('yayyyyy we did it')
  console.log(response)
  resetForm($('#delete-chore'))
  console.log('Is the form empty now?')
  $('#deleteChoreSuccessPrompt').text('You have deleted a chore!')
}

const deleteChoreFailure = (response) => {
  console.log('boooo it failed')
  console.log(response)
  $('#deleteChoreFailurePrompt').text('Deleting a chore failed. Try again.')
}
```

I feel like I should be doing something more in events.js to be sure that (id) is actually what I want, but I'm not sure how to get into the object.

I read a few other issues, but they didn't seem quite relevant to this.  Pointers appreciated. Thank you!

**Comment by danman01** [comment]
You have your code well set up to debug this:
```
getFormFields(event.target)
```
is logged in your console. What do you think `[Object object]` is, and what are you actually trying to pass on to your API call?

**Comment by lisawilliams** [comment]
I'm pretty sure it's passing the chore object returned by the console.log(id) statement. What I want to pass on is just the chore ID, in this case, '5'. 

I *think* the resulting url should look like: 

`http://localhost:4741/chores/5`

**Comment by danman01** [comment]
That sounds correct. So, from `event.target` (which is a js object), how can you access just the ID?

**Comment by lisawilliams** [comment]
I feel like I should do: 

`  const id = getFormFields(event.target).attr(id)`

But when I do that, it fails, and I get a kind of unhelpful error thrown on that line. 

`TypeError: getFormFields(...).attr is `

...and then that's it, it doesn't say what it is.

**Comment by danman01** [comment]
Do you know the structure of the object returned from `event.target`? If not, how can you investigate what is contained within that object so that you can access a property of the object?

**Comment by lisawilliams** [comment]
OK, so: What's returned by (id) is the entire chore object.  

So I declared ID as a variable, and it now has an object within it that looks like this: 

```
{ "chore": 
    { "id": 5
        }
}
```

So what we want is not the variable `id` but to get at an object property within it, in this case:

`id.chore.id` 

Which returns '5', as expected (that's what I typed into the form). (The naming of "id" for the variable was pretty confusing and I may change it). 

When I changed my code to read as follows: 

```
const onDeleteChore = function (event) {
  event.preventDefault()
  const id = getFormFields(event.target)
  console.log('we have arrived at onDeleteChore chores/events.js. Below is console.log event.target')
  console.log(event.target)
  // debugger;
  console.log('Below is console.log(id)')
  console.log(id.chore.id)
  api.deleteChore(id.chore.id)
  .then(ui.deleteChoreSuccess)
  .catch(ui.deleteChoreFailure)
}
```
Note especially api.deleteChore(id.chore.id).  Now it is passing '5' which means that the URL being passed to the API telling it which chore to delete is `http://localhost:4741/chores/5`

This DELETE request runs, and returns `Completed 204 No Content` as expected in the Rails Console. Checking the `chores` table by running `bin/rails db` and then TABLE chores; shows that the chore with the chore id of 5 is now gone. Huzzah!
